server/src/domain/asset/asset.service.ts

```diff
@@ -159,14 +161,19 @@ export class AssetService {
 
   async getMapMarkers(auth: AuthDto, options: MapMarkerDto): Promise<MapMarkerResponseDto[]> {
     const userIds: string[] = [auth.user.id];
+    const albumIds: string[] = [];
     if (options.withPartners) {
       const partners = await this.partnerRepository.getAll(auth.user.id);
       const partnersIds = partners
         .filter((partner) => partner.sharedBy && partner.sharedWith && partner.sharedById != auth.user.id)
         .map((partner) => partner.sharedById);
       userIds.push(...partnersIds);
     }
-    return this.assetRepository.getMapMarkers(userIds, options);
+    if (options.withSharedAlbums) {
+      const sharedAlbums = await this.albumRepository.getShared(auth.user.id);
```

Comment:
This doesn't include the asset relation does it?

Reply:
I don't follow, where should I include the asset relation?

Reply:
Okay so Jason answered me on Discord about this:
https://discord.com/channels/979116623879368755/994044917355663450/1238973376349864108

Jason said: "Just want to make sure this is not loading the album AND all the assets for it"

I said: "If I understand correctly, you are asking if it is loading only the shared albums when figuring out what to show on the map, versus loding the shared albums and all the assets. And the problem with that is loading all the assets in the shared album takes more resources."

Jason responded: "Yup"

Reply:
yes, this function does not query all assets, only the albums:
https://github.com/immich-app/immich/blob/f667c9597b9885059de00d9e8779147c4d08ddc2/server/src/repositories/album.repository.ts#L154-L166